firewood/src/merkle/proof.rs

```diff
@@ -330,16 +333,14 @@ impl<N: AsRef<[u8]> + Send> Proof<N> {
                 _ => return Err(ProofError::InvalidNode(MerkleError::ParentLeafBranch)),
             };
 
-            u_ref = merkle.get_node(chd_ptr)?;
+            node_ref = merkle.get_node(chd_ptr)?;
 
             // If the new parent is a branch node, record the index to correctly link the next child to it.
-            if u_ref.inner().as_branch().is_some() {
-                #[allow(clippy::indexing_slicing)]
-                (branch_index = chunks[key_index] as usize);
+            #[allow(clippy::unwrap_used)]
+            if node_ref.inner().as_branch().is_some() {
+                child_index = key_nibbles.next().unwrap() as usize;
```

Comment:
New code should not use `unwrap()`.

It's not clear why this unwrap is safe. Why are we sure that there is another nibble at this point?

Reply:
Yeah, this needs to be fixed. Working on it

Reply:
fixed